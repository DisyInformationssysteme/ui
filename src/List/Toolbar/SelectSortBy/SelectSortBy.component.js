import React, { PropTypes } from 'react';
import { Button, Nav, NavDropdown, MenuItem } from 'react-bootstrap';
import Icon from '../../../Icon';

/**
 * @param {object} props react props
 * @example
 <SelectSortBy name="Hello world"></SelectSortBy>
 */
function SelectSortBy(props) {
	let selected;
	props.sortBy.forEach((sortBy) => {
		if (sortBy.selected) {
			selected = sortBy;
		}
	});
	let onSelectSortBy = props.onSelectSortBy;
	if (onSelectSortBy) {
		onSelectSortBy = (key, event) => props.onSelectSortBy(
			event,
			{
				sortBy: key.id,
				sortDesc: !!props.sortDesc,
			}
		);
	}
	const toggleSortOrder = () => {
		if (props.onSelectSortBy) {
			props.onSelectSortBy(
				event,
				{
					sortBy: selected.id,
					sortDesc: !props.sortDesc,
				}
			);
		}
	};
	return (
		<Nav>
			<label className="navbar-text" htmlFor="tc-list-toolbar-sort-by">Sort by:</label>
			<NavDropdown
				title={selected ? selected.name : 'N.C'}
				id="tc-list-toolbar-sort-by"
				onSelect={onSelectSortBy}
			>
				{props.sortBy.map((sortBy, index) => (
					<MenuItem key={index} eventKey={sortBy}>{sortBy.name}</MenuItem>
				))}
			</NavDropdown>
			<Button className="navbar-btn" bsStyle="link" role="button" onClick={toggleSortOrder}>
				<Icon name={props.sortDesc ? 'fa-sort-desc' : 'fa-sort-asc'} />
				{props.sortDesc ? 'DESCENDING' : 'ASCENDING'}
			</Button>
		</Nav>
	);
}

SelectSortBy.propTypes = {
	sortBy: PropTypes.arrayOf(
		PropTypes.shape({
			id: PropTypes.string,
			name: PropTypes.string,
			selected: PropTypes.bool,
		})
	),
	onSelectSortBy: PropTypes.func,
	sortDesc: PropTypes.bool,
};

export default SelectSortBy;
