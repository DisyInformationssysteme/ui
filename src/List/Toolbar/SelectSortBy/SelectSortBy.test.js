import React from 'react';
import renderer from 'react-test-renderer';

import SelectSortBy from './SelectSortBy.component';

jest.mock('react-dom');

describe('SelectSortBy', () => {
	it('should render', () => {
		const props = {
			onSelectSortBy: jest.fn(),
			sortBy: [
				{ id: 'id', name: 'Name', selected: true },
				{ id: 'name', name: 'Name' },
			],
		};
		const wrapper = renderer.create(
			<SelectSortBy {...props} />
		).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
	it('should render without sortBy selected', () => {
		const props = {
			onSelectSortBy: jest.fn(),
			sortBy: [
				{ id: 'id', name: 'Name' },
				{ id: 'label', name: 'Label' },
			],
		};
		const wrapper = renderer.create(
			<SelectSortBy {...props} />
		).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
