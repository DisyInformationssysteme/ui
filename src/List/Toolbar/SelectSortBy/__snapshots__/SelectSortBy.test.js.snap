exports[`SelectSortBy should render 1`] = `
<ul
  className="nav"
  role={null}>
  <label
    active={undefined}
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-text"
    htmlFor="tc-list-toolbar-sort-by"
    onSelect={null}>
    Sort by:
  </label>
  <li
    className="dropdown"
    style={undefined}>
    <a
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle"
      href=""
      id="tc-list-toolbar-sort-by"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button">
      Name
       
      <span
        className="caret" />
    </a>
    <ul
      aria-labelledby="tc-list-toolbar-sort-by"
      className="dropdown-menu"
      role="menu">
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          Name
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          Name
        </a>
      </li>
    </ul>
  </li>
  <button
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-btn btn btn-link"
    disabled={false}
    onClick={[Function]}
    onSelect={null}
    role="button"
    type="button">
    <i
      aria-hidden="true"
      className="fa fa-sort-asc"
      title={null} />
    ASCENDING
  </button>
</ul>
`;

exports[`SelectSortBy should render without sortBy selected 1`] = `
<ul
  className="nav"
  role={null}>
  <label
    active={undefined}
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-text"
    htmlFor="tc-list-toolbar-sort-by"
    onSelect={null}>
    Sort by:
  </label>
  <li
    className="dropdown"
    style={undefined}>
    <a
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle"
      href=""
      id="tc-list-toolbar-sort-by"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button">
      N.C
       
      <span
        className="caret" />
    </a>
    <ul
      aria-labelledby="tc-list-toolbar-sort-by"
      className="dropdown-menu"
      role="menu">
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          Name
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          Label
        </a>
      </li>
    </ul>
  </li>
  <button
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-btn btn btn-link"
    disabled={false}
    onClick={[Function]}
    onSelect={null}
    role="button"
    type="button">
    <i
      aria-hidden="true"
      className="fa fa-sort-asc"
      title={null} />
    ASCENDING
  </button>
</ul>
`;
